src/Components/Forms/Inputs/Select.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BladeUIKit\Components\Forms\Inputs;
+
+use BladeUIKit\Components\BladeComponent;
+use http\Exception\InvalidArgumentException;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Collection;
+
+class Select extends BladeComponent
+{
+    /** @var string */
+    public $name;
+
+    /** @var string|null */
+    public $id;
+
+    /** @var array|Collection */
+    public $options;
+
+    /** @var string */
+    public $nameAttribute;
+
+    /** @var string */
+    public $valueAttribute;
+
+    /** @var string|array|null */
+    public $selected;
+
+    /** @var string */
+    public $placeholder;
+
+    public function __construct(string $name, $options, string $placeholder = '', string $nameAttribute = 'name', string $valueAttribute = 'id', $id = null, $selected = null)
+    {
+        $this->name = $name;
+        $this->id = $id ?? $name;
+        $this->nameAttribute = $nameAttribute;
+        $this->valueAttribute = $valueAttribute;
+        $this->selected = $selected;
+        $this->placeholder = $placeholder;
+
+        if ($options instanceof Collection) {
+            $this->options = $options->mapWithKeys(function ($option) {
```

Comment:
We can refactor this `mapWithKeys()` expression to use `pluck()`

```suggestion
            $this->options = $options->pluck($this->nameAttribute, $this->valueAttribute)
```